conformance/utils/suite/features.go

```diff
@@ -0,0 +1,163 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package suite
+
+import "k8s.io/apimachinery/pkg/util/sets"
+
+// -----------------------------------------------------------------------------
+// Features - Types
+// -----------------------------------------------------------------------------
+
+// SupportedFeature allows opting in to additional conformance tests at an
+// individual feature granularity.
+type SupportedFeature string
+
+// -----------------------------------------------------------------------------
+// Features - Standard (Core)
+// -----------------------------------------------------------------------------
+
+const (
+	// This option indicates support for ReferenceGrant (core conformance).
+	// Opting out of this requires an implementation to have clearly implemented
+	// and documented equivalent safeguards.
+	SupportReferenceGrant SupportedFeature = "ReferenceGrant"
+)
+
+// StandardCoreFeatures are the features that are required to be conformant with
+// the Core API features (e.g. GatewayClass, Gateway, e.t.c.).
+//
+// TODO: rename this from "Standard" as that might be confusing with our
+// "Standard" release channel.
+var StandardCoreFeatures = sets.New(
+	SupportReferenceGrant,
+)
+
+// -----------------------------------------------------------------------------
+// Features - Standard (Extended)
+// -----------------------------------------------------------------------------
+
+const (
+	// This option indicates GatewayClass will update the observedGeneration in
+	// it's conditions when reconciling (extended conformance).
+	//
+	// TODO: we intend to make this core and require implementations to do it
+	//       as we expect this is something every implementation should be able
+	//       to do and it's ideal behavior.
+	//
+	//       See: https://github.com/kubernetes-sigs/gateway-api/issues/1780
```

Comment:
Maybe just make this a note instead of a TODO and then add some kind of issue to track removing this?

Reply:
I guess the existing issue is fine here, I'd just prefer to replace "TODO:" with "Note:".

Reply:
Ah, I gotcha now. Changed!